css-typed-om/Overview.bs

```diff
@@ -784,10 +782,29 @@ The following are the arithmetic operations you can perform on dimensions:
         is set to |this|.
 </div>
 
-<div class=issue>
-    Define {{CSSNumericValue/equals()}}.
+<div algorithm="CSSNumericValue.equals(value)">
+    The <dfn for=CSSNumericValue method>equals(...|values|)</dfn> method,
+    when called on a {{CSSNumericValue}} |this|,
+    must perform the following steps:
+
+    1. Replace each [=list/item=] of |values|
+        with the result of [=rectifying a numberish value=] for the [=list/item=].
+
+    2. Let |thisAndValues| be |this| prepended to |values|.
+
+    1. [=list/If all=] |item|s of |thisAndValues| do not belong to the same class return false. Example: all items in |thisAndValues| must be {{CSSUnitValue}}.
+
+    2. [=list/If all=] |item|s in |thisAndValues| are {{CSSUnitValue}s and have the same {{CSSUnitValue/value}} return true.
+
+    4. =list/If all=] |item|s in |thisAndValues| are {{CSSMathValue}s :
+        1. If the {{CSSMathValue/operator}} are not equal for all {{CSSMathValues}} in |thisAndValues| return false.
+        2. If |thisAndValues| are all either {{CSSMathSum}}, {{CSSMathProduct}}, {{CSSMathMin}} or {{CSSMathMax}}, iterate over the {{CSSNumericArray}}s on |thisAndValues|,
+        	1. For each {{CSSUnitValue}} in each of the {{CSSNumericArray}}s go back to step 1 to check that the {{CSSUnitValue}}s on all items in |thisAndValue| are equal.
+    	3. If |this| and |value| are both either {{CSSMathNegate}} or {{CSSMathInvert}}:
```

Comment:
Do you mean CSSMathProduct? Cause that is dealt with in the above step. Or do you mean the case where values in |thisAndValues| are not all of the same type? If the latter in that case we should return a false right?